src/snowflake/snowpark/_internal/data_source/jdbc.py

```diff
@@ -0,0 +1,582 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+import logging
+import re
+from collections import defaultdict
+from enum import Enum
+from functools import cached_property
+from typing import Optional, Union, List, TYPE_CHECKING
+
+from snowflake.snowpark._internal.data_source import DataSourcePartitioner
+from snowflake.snowpark._internal.data_source.drivers import BaseDriver
+from snowflake.snowpark._internal.utils import (
+    generate_random_alphanumeric,
+    random_name_for_temp_object,
+    TempObjectType,
+)
+from snowflake.snowpark.types import (
+    StructType,
+    DecimalType,
+    FloatType,
+    StringType,
+    DateType,
+    BooleanType,
+    ArrayType,
+    StructField,
+    VariantType,
+    TimestampType,
+    TimestampTimeZone,
+    IntegerType,
+    DoubleType,
+    TimeType,
+)
+
+if TYPE_CHECKING:
+    from snowflake.snowpark.session import Session
+    from snowflake.snowpark.dataframe import DataFrame
+
+logger = logging.getLogger(__name__)
+
+# reference for java type to snowflake type:
+# https://docs.snowflake.com/en/developer-guide/udf-stored-procedure-data-type-mapping
+
+
+class JDBCType(Enum):
+    # ARRAY = "ARRAY"  # not supported in snowflake
```

Comment:
let's create a JIRA as followups on these unsupported types? I feel some types can be supported with additional efforts (BLOB -> bytes -> hexstring -> snowflake binary)